### PR TITLE
feat: webhook adapter for next.js api routes

### PR DESCRIPTION
**Pull request by waptik**
This adapter will enable bots to work smoothly when using next.js api routes as webhook endpoint. See https://t.me/grammyjs/86659 for how it started.

### CODE REVIEW

File: src/convenience/frameworks.node.ts [hunk 1]
```diff
@@ -67,13 +66,23 @@ const azure = (context: any, req: any) => ({
     },
 });
 
+/** Vercel/Next.js Serverless Functions */
+const vercel = (req: any, res: any) => ({
+    update: Promise.resolve(req.body),
+    header: req.headers[SECRET_HEADER] as string,
```
Comment (KnorpelSenf) [inline src/convenience/frameworks.node.ts:72]:
```suggestion
    header: req.headers[SECRET_HEADER],
```

File: src/convenience/frameworks.node.ts [hunk 2]
```diff
@@ -53,11 +53,10 @@ const awsLambda = (event: any, _context: any, callback: any) => ({
 const azure = (context: any, req: any) => ({
     update: Promise.resolve(req.body),
     header: context.res.headers[SECRET_HEADER],
-    end: () =>
-        context.res = {
-            status: 200,
-            body: "",
-        },
+    end: () => (context.res = {
```
Comment (KnorpelSenf) [inline src/convenience/frameworks.node.ts:56]: This change is useless and only makes the code harder to read, please revert

File: src/convenience/frameworks.node.ts [hunk 3]
```diff
@@ -67,13 +66,23 @@ const azure = (context: any, req: any) => ({
...
+});
+
 // please open a PR if you want to add another
 export const adapters = {
     http,
     https: http,
     worktop,
     "aws-lambda": awsLambda,
     azure,
+    vercel,
```
Comment (KnorpelSenf) [inline src/convenience/frameworks.node.ts:85]: This name is misleading. Not all bots hosted on Vercel should use this adapter.

Is it possible to use this adapter when using Next.js outside of Vercel?

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]

File: src/convenience/frameworks.node.ts (hunk 2 above)
Comment (waptik) [reply src/convenience/frameworks.node.ts:56]: It was generated after running `deno fmt` but sure.

File: src/convenience/frameworks.node.ts (hunk 3 above)
Comment (waptik) [reply src/convenience/frameworks.node.ts:85]: Yes, it's possible. So should i name it `next.js` or `next` or `nextjs`?

File: src/convenience/frameworks.node.ts (hunk 2 above)
Comment (KnorpelSenf) [reply src/convenience/frameworks.node.ts:56]: Oh

File: src/convenience/frameworks.node.ts (hunk 3 above)
Comment (KnorpelSenf) [reply src/convenience/frameworks.node.ts:85]: Either `nextjs` or `next.js` or `next-js` is fine by me. Just `next` would be confusing. Pick whichever one you prefer.

File: src/convenience/frameworks.node.ts (hunk 2 above)
Comment (waptik) [reply src/convenience/frameworks.node.ts]: So what should i do? It still formats after i save the file

Comment (KnorpelSenf) [reply src/convenience/frameworks.node.ts:56]: Then we can perform the changes that way. I'll try to reproduce it and then open a bug report for the Deno formatter. Don't worry about it for now.

**Review by KnorpelSenf — APPROVED** [review]
Thank you!

File: src/convenience/frameworks.node.ts (hunk 2 above)
Comment (waptik) [reply src/convenience/frameworks.node.ts:56]: Okay, thanks!

**Pull request merged**